### DISCUSSION TITLE
Add Tidal support to Music providers

### DISCUSSION THREAD

**Discussion by marcelveldt** [music-assistant/support#432] [Feature requests and ideas]
Add support for Tidal as music source/provider

**Comment by ezpl** [comment]
I would love to help with testing this as soon as there is anything that could possibly connect.

**Comment by nzachow** [comment]
If someone can provide me some pointers on where to start implementing this feature I can try to code this integration.

**Reply by marcelveldt** [reply]
please contact me on discord: marcelveldt#0393

**Reply by S1M8N** [reply]
Have you news for this feature ?

**Reply by marcelveldt** [reply]
It's being worked on by a nice community member

**Reply by MadsTheEngineer** [reply]
That is awesome! What is the ETA for a feature like this?

**Reply by quizzical86** [reply]
Very glad to read that there's a community member working on this (and a nice bonus that it's a nice one :-D )
To whomever it is: thanks!!!

**Reply by majbom** [reply]
Good to hear. I don't hope your new job situation gets in the way.
Congratulations by the way :)

**Reply by Eric-N-Be** [reply]
realy nice idee ... thanks at the community ...  I'd be happy to be a tester

**Comment by jozefKruszynski** [comment]
If you happen to need another tester, please count me in

**Comment by EpicMount** [comment]
I'd be happy to be a tester as well. FWIW, I have a Tidal Premium account.

**Comment by sebrep** [comment]
where is the code, I could maybe help out a bit, is there a branch?

**Reply by marcelveldt** [reply]
@nzachow do you have the code in a branch somewhere so people can jump in and help ?

**Reply by hallidaydaniel** [reply]
Any news on this, is there a branched / forked repo to look at and help out with?

**Reply by CoreyJ87** [reply]
wondering the same. I would like to get this going if possible.

**Reply by quizzical86** [reply]
Dear @sebrep, @hallidaydaniel  and @CoreyJ87 thanks a lot for offering your help in making this happen. It's by far the most upvoted feature request for Music Assistant. I was wondering whether @marcelveldt and @nzachow have taken you up on your offer?
I would love this feature because the native Tidal integration with my Denon HEOS speakers is quite poor (e.g it can't play the automatically generated playlists based on my music taste). 
Thanks for your insights!

**Comment by oneofthemany** [comment]
cant wait for this feature - if any testers are needed give me a shout

**Comment by JimmyMPage** [comment]
This is great news! I love Tidal. Can't wait. Willing to test as well.

**Comment by gogui63** [comment]
@marcelveldt If necessary, I can add you to my premium account (Hifi family account with free slots).

**Comment by CoreyJ87** [comment]
Any updates on this? I really wanna use MA but I'm waiting on tidal support. Since that all I have.

**Comment by Jarvisv12** [comment]
Can't wait would love this

**Comment by Tontze** [comment]
Awesome 👯‍♂️

**Comment by marcelveldt** [comment]
I will try to include Tidal in the upcoming v2.0 release of MA and if that doesn't make it, it will be in the follow up release shortly after. So, a little bit more patience of a few weeks please.

**Reply by quizzical86** [reply]
Bedankt voor de update Marcel, much appreciated!

**Comment by grzywek** [comment]
Would that be able to play music via MA but still control it from Tidal app? I have media player connected to Home Assistant which is not visible in the Tidal app itself.

**Reply by Madelena** [reply]
Yeah, it would be nice if it works as a Tidal Connect speaker.

**Comment by mariuszpelka** [comment]
@marcelveldt anything regarding Tidal integration in MA? We are in half march now :) 
Looking at commit activity the project is frozen. Tidal connect or any Tidal integration still invisible in code.

**Reply by OzGav** [reply]
It is in the queue but lowest priority at this time

**Comment by oneofthemany** [comment]
Are you guys integrating this API:

https://pypi.org/project/tidalapi/

Seems like some of the leg work has been done

**Comment by OzGav** [comment]
There is discussion occurring about this on discord. So hopefully someone will pick this up soon.

**Comment by Sirs0ri** [comment]
There's been some progress and Tidal's been in the 2.0.0 beta builds for a while, see #1201!

**Comment by OzGav** [comment]
See here https://github.com/orgs/music-assistant/discussions/1201
